### DISCUSSION TITLE
Lesson 4: Facing Problem while installing web3 using pip

### DISCUSSION THREAD

**Discussion by Noveleader** [smartcontractkit/full-blockchain-solidity-course-py#1317] [Q&A] [answered]
I am getting following error message after installing web3 on my system. I also installed Microsoft Visual C++ but isn't helping a lot maybe I did something wrong. 
```
× Running setup.py install for lru-dict did not run successfully.  │ exit code: 1
  ╰─> [5 lines of output]
      running install
      running build
      running build_ext
      building 'lru' extension
      error: Microsoft Visual C++ 14.0 or greater is required. Get it with "Microsoft C++ Build Tools": https://visualstudio.microsoft.com/visual-cpp-build-tools/  
      [end of output]

  note: This error originates from a subprocess, and is likely not a problem with pip.
error: legacy-install-failure

× Encountered error while trying to install package.
╰─> lru-dict

note: This is an issue with the package mentioned above, not pip.
hint: See above for output from the failure.
```
Help me out please!!

**Comment by cromewar** [comment]
Hello @Noveleader please refer to: https://github.com/smartcontractkit/full-blockchain-solidity-course-py/blob/main/chronological-issues-from-video.md#python-related-issues
This is a guide about how to solve web3 problems, I hope this might help.

**Reply by Noveleader** [reply]
Hey there is some problem with my system I feel I don't have Visual but even after installing it, I am still facing same issue but since web 3 is installed I will continue the course for now.

**Reply by cromewar** [reply]
You can try with WSL using the guide "windows support" on the course index.

**Reply by Noveleader** [reply]
> You can try with WSL using the guide "windows support" on the course index.

Great, I will do that

**Comment by Noveleader** [comment, marked as answer]
I have solved this problem, there is some problem with the version of Visual C++ which you can find [here](https://www.microsoft.com/en-in/download/details.aspx?id=48145) and after updating it run the installation process again and this should solve the problem. Even after this problem remains download Build tools for Visual Studio 2022 under Tools for Visual Studio 2022 [here](https://visualstudio.microsoft.com/downloads/)

**Reply by formulaRoot** [reply]
Is this solution going to work for linux as well or is it only windows related

**Reply by Noveleader** [reply]
It is windows related

**Reply by formulaRoot** [reply]
Thanks @Noveleader  i figured as much. 

I couldnt solve it on linux but its running fine on Mac right now.

**Reply by Noveleader** [reply]
ohh great
